bundles/org.openhab.core.io.rest.swagger/src/main/java/org/openhab/core/io/rest/swagger/impl/SwaggerResource.java

```diff
@@ -23,32 +23,36 @@
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.core.io.rest.RESTConstants;
+import org.openhab.core.io.rest.RESTResource;
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
 import org.osgi.service.jaxrs.whiteboard.propertytypes.JSONRequired;
 import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsApplicationSelect;
+import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsName;
 import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsResource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import de.maggu2810.jaxrswb.swagger1.gen.JaxRsWhiteboardSwaggerGenerator;
 
 /**
- * An endpoint to generate and provide a Swagger 1 description.
+ * An endpoint to generate and provide a Swagger description.
  *
  * @author Markus Rathgeb - Initial contribution
+ * @author Kai Kreuzer - made it a RESTResource to register in the root bean
  */
-@Component(service = RESTResource.class)
+@Component(service = SwaggerResource.class)
 @JaxrsResource
+@JaxrsName("spec")
 @JaxrsApplicationSelect("(" + JaxrsWhiteboardConstants.JAX_RS_NAME + "=" + RESTConstants.JAX_RS_NAME + ")")
 @JSONRequired
-@Path("/swagger.json")
+@Path("/spec")
```

Comment:
In https://github.com/openhab/openhab-webui/pull/297#discussion_r459763420 you reference a different path?

Reply:
Haha, good catch - not anymore now ;-)